tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_ops.py

```diff
@@ -54,9 +50,49 @@
 from tensorflow.python.util import compat
 from tensorflow.python.util.tf_export import tf_export
 
+from tensorflow.python.keras.utils import tf_utils
+from tensorflow.python.distribute import distribution_strategy_context as distribute_ctx
+from tensorflow.python.distribute import values as distribute_values_lib
+from tensorflow.python.training.saving import saveable_object
+from tensorflow.python.distribute import distribute_utils
+
 _ANONYMOUS_TRAINABLE_STORE_KEY = '_anonymous_trainable_store_key'
 
 
+class TrainableWrapperDistributedPolicy(distribute_values_lib.OnWritePolicy):
+
+  def get_saveable(self, var, primary_var, name):
+    for v in var.values:
+      v._gather_saveables_for_checkpoint()
+
+    def tensor():
+      return array_ops.zeros((
+          0,
+          primary_var.params.dim,
+      ),
+                             dtype=primary_var.params.value_dtype)  # pylint: disable=protected-access
+
+    spec = saveable_object.SaveSpec(tensor=tensor,
+                                    slice_spec="",
+                                    name=name,
+                                    dtype=primary_var.params.value_dtype,
+                                    device=primary_var.device)
+    return tensor, [spec]
+
+
+class DistributedVariableWrapper(distribute_values_lib.DistributedVariable):
```

Comment:
What's the value of the class? It looks similar to the superclass?
Try this?
`DistributedVariableWrapper = distribute_values_lib.DistributedVariable`

Reply:
This allows TFRA to individually recognize its own DistributedVariable and process it separately.  Reference to 
```python
def DistributedVariableWrapperFilter(grads_and_vars_in):
```
https://github.com/MoFHeka/recommenders-addons/blob/cefa36a9daf37be129355b1bf70cb6f2e6143421/tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_optimizer.py#L688